app/controllers/user_actions_controller.rb

```diff
@@ -2,8 +2,7 @@
 
 class UserActionsController < ApplicationController
   def index
-    params.require(:username)
-    params.permit(:filter, :offset, :acting_username, :limit)
```

Comment:
This line wasn’t doing anything as `#permit` returns a new object and doesn’t mutate the one in place.

Reply:
I thought the `permit` was used to allowlist fields of the `params` object. So it generates an exception when you try to access fields that weren't permitted?

Reply:
No actually only `require` raises an exception, `permit` will only filter the arguments and return a new `ActionController::Parameters` object. But re-reading the docs, it seems it should not authorize hashes in its filtering :thinking: Going to re-check that so maybe we could simply use the result of `#permit` instead of having to cast all values to strings.

Reply:
>  Going to re-check that so maybe we could simply use the result of #permit instead of having to cast all values to strings.

I would very much prefer that to the `transform_keys!` hack

Reply:
Indeed, `#permit` is filtering hashes. The docs say it will allow those value types: `String`, `Symbol`, `NilClass`, `Numeric`, `TrueClass`, `FalseClass`, `Date`, `Time`, `DateTime`, `StringIO`, `IO`, `ActionDispatch::Http::UploadedFile` or `Rack::Test::UploadedFile`. So I guess if we get something else than a string for whatever reason we will still end up with the same problem though. But let’s see how it goes with that approach first I’d say.

Reply:
Done! :slightly_smiling_face: